lib/strong_migrations/migration_helpers.rb

```diff
@@ -0,0 +1,70 @@
+module StrongMigrations
+  module MigrationHelpers
+
+    # Adds a foreign key with minimal locking on the tables involved.
+    #
+    # Example:
+    #
+    #     add_foreign_key_concurrently :articles, :authors
+    #
+    # See Rails' `add_foreign_key` for more info on the available options.
+    def add_foreign_key_concurrently(from_table, to_table, options = {})
+      ensure_not_in_transaction(__method__)
+
+      if ActiveRecord::VERSION::STRING >= "5.2"
+        add_foreign_key(from_table, to_table, options.merge(validate: false))
+        validate_foreign_key(from_table, to_table)
+      else
+        options = foreign_key_options(from_table, to_table, options)
+
+        # Using NOT VALID constraint option allows to reduce the impact of adding
+        # a constraint on concurrent updates. With NOT VALID, the ADD CONSTRAINT
+        # command does not scan the table and can be committed immediately. Added
+        # constraint is enforced for any newly created data.
+        safety_assured do
+          execute <<~SQL
+            ALTER TABLE #{from_table}
```

Comment:
Identifiers need to be quoted here and with the `VALIDATE CONSTRAINT` command - see the `constraint_str` method.